maven/test-framework/org.jboss.tools.maven.reddeer/src/org/jboss/tools/maven/reddeer/wizards/MavenProjectWizardSecondPage.java

```diff
@@ -11,27 +11,52 @@
 package org.jboss.tools.maven.reddeer.wizards;
 
 import org.eclipse.reddeer.jface.wizard.WizardPage;
+import org.eclipse.reddeer.swt.api.TableItem;
 import org.eclipse.reddeer.swt.condition.TableHasRows;
 import org.eclipse.reddeer.swt.impl.combo.DefaultCombo;
 import org.eclipse.reddeer.swt.impl.table.DefaultTable;
+import org.eclipse.reddeer.swt.impl.table.DefaultTableItem;
 import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
 import org.eclipse.reddeer.common.wait.TimePeriod;
 import org.eclipse.reddeer.common.wait.WaitUntil;
 import org.eclipse.reddeer.common.wait.WaitWhile;
 import org.eclipse.reddeer.core.reference.ReferencedComposite;
 
+import java.util.List;
+
 public class MavenProjectWizardSecondPage extends WizardPage{
 	
 	public MavenProjectWizardSecondPage(ReferencedComposite referencedComposite) {
 		super(referencedComposite);
 	}
+	
+	public void selectArchetype(String catalog, final String archetype){
+		selectArchetype(catalog, archetype, null);
+	}
 
-	public void selectArchetype(String catalog, String archetype){
+	public void selectArchetype(String catalog, final String archetype, final String groupId){
 		new DefaultCombo(0).setSelection(catalog);
 		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
 		new WaitUntil(new TableHasRows(new DefaultTable()),TimePeriod.LONG);
-		// TODO: RedDeer has to be enhanced to find item by text of other rows then just first one
-		new DefaultTable().getItem(archetype,1).select();
+		DefaultTable table = new DefaultTable();
+				 	 	    
+	    @SuppressWarnings("unchecked")
+		List<TableItem> items = table.getItems(new Matcher<TableItem>() {
+			@Override
+			public void describeTo(Description arg0) {}
+			@Override
+			public void _dont_implement_Matcher___instead_extend_BaseMatcher_() {}
```

Comment:
Maybe we should follow the recommandation, no ?

Reply:
Ah, sure, that forced me to look for the answer why would anybody make a [comment as a method name](https://stackoverflow.com/questions/20712782/why-insist-all-implementations-of-an-interface-extend-a-base-class). I was trying to avoid adding more separated classes but if that is desired result what package should contains this new ArtifactIdAndGroupIdMatcher extended from BaseMatcher? I would like to suggest org.jboss.tools.maven.ui.bot.test/src/org/jboss/tools/maven/ui/bot/test/project/matchers/ArtifactIdAndGroupIdMatcher.java . Any other ideas?

Reply:
Ah, sure, that forced me to look for the answer why would anybody make a [comment as a method name](https://stackoverflow.com/questions/20712782/why-insist-all-implementations-of-an-interface-extend-a-base-class). I was trying to avoid adding more separated classes but if that is desired result what package should contains this new ArtifactIdAndGroupIdMatcher extended from BaseMatcher? I would like to suggest org.jboss.tools.maven.ui.bot.test/src/org/jboss/tools/maven/ui/bot/test/project/matchers/ArtifactIdAndGroupIdMatcher.java . Any other ideas?

Reply:
You can extend a class the same way when creating a anonymous class. But I'm fine with the proposed name